src/platform_linux/platform.h

```diff
@@ -627,15 +627,15 @@ void
 platform_heap_destroy(platform_heap_handle *heap_handle);
 
 buffer_handle *
-platform_buffer_create(size_t               length,
-                       platform_heap_handle heap_handle,
-                       platform_module_id   module_id);
+platform_buffer_create(size_t             length,
+                       platform_heap_id   heap_id,
+                       platform_module_id module_id);
 
 void *
 platform_buffer_getaddr(const buffer_handle *bh);
 
 platform_status
-platform_buffer_destroy(buffer_handle *bh);
+platform_buffer_destroy(platform_heap_id heap_id, buffer_handle **bh_in);
```

Comment:
Can you just add a field to the buffer handle so it remembers its own heapid so you don't need to pass the heapid to the destroy function?

Reply:
This comment is obviated by my high level proposal.  I think the high-level proposal will eliminate the need for a heap parameter altogether.

Reply:
Like you observed, elsewhere, this new parameter `heap_id` became unnecessary for these interfaces. I didn't have to add it to the `buffer_handle` struct either, which keeps this change simpler.